File: Leetcode/0190_reverse_bits/0190_reverse_bits.cs
public class Solution {
    public uint reverseBits(uint n) {
        uint result = 0;
        for (int i = 0; i < 32; i++) {
            result = (result << 1) | (n & 1); // Shift result to the left and add the least significant bit of n
            n >>= 1; // Shift n to the right
        }
        return result;
    }
}

File: Leetcode_past/algorithms/easy/0190_reverse_bits/0190_reverse_bits.go
func reverseBits(num uint32) uint32 {
    var result uint32 = 0
    var i uint = 0
    
    for i < 32 {
        result = (result << 1) | (num & 1) // Shift result to the left and add the least significant bit of num
        num >>= 1 // Shift num to the right
        i++
    }
    
    return result
}
